### ISSUE TITLE
SELECT AS <quoted keyword> rejected

### ISSUE THREAD

**Issue by agentm** [JakeWheat/simple-sql-parser#22]
`SELECT x AS "period";` fails to parse because "period" is a blacklisted keyword despite the fact that it is quoted. 

Does SQL2011 actually prohibit quotes identifiers from appearing in the result list?

**Comment by JakeWheat** [comment]
Thanks for the report, this is definitely a bug in the parsing code!

**Comment by agentm** [comment]
Does this patch make sense?

```
diff --git a/Language/SQL/SimpleSQL/Parse.lhs b/Language/SQL/SimpleSQL/Parse.lhs
index 397e313..91eb2b5 100644
--- a/Language/SQL/SimpleSQL/Parse.lhs
+++ b/Language/SQL/SimpleSQL/Parse.lhs
@@ -2138,6 +2138,7 @@ It is only allowed when all the strings are quoted with ' atm.
 > identifierTok :: [String] -> Parser (Maybe (String,String), String)
 > identifierTok blackList = mytoken (\tok ->
 >     case tok of
+>       L.Identifier q@(Just ("\"","\"")) p -> Just (q,p)
 >       L.Identifier q p | map toLower p `notElem` blackList -> Just (q,p)
 >       _ -> Nothing)
 
```

**Comment by JakeWheat** [comment]
Yes, that looks great. I would just change the pattern from `q@(Just ("\"","\""))` to `q@(Just {})`, since I think it should be fine whatever quoting is used, e.g. sql server style quoting - `select * from [period]`.

**Comment by agentm** [comment]
I see- I thought you might wish to guard the quoting by dialect.

**Comment by JakeWheat** [comment]
fixed in 0.6.0 release
